docs/model-dev-guide/dtrain/dtrain-introduction.rst

```diff
@@ -0,0 +1,278 @@
+ Distributed Training Concepts
+###############################
+
+*******************************************
+ How Determined Distributed Training Works
+*******************************************
+
+Distributed training in Determined utilizes data-parallelism. Data-parallelism for deep-learning
+consists of a set of workers, where each worker is assigned to a unique compute accelerator such as
+a GPU or a TPU. Each worker maintains a copy of the model parameters (weights that are being
+trained), which is synchronized across all the workers at the start of training.
+
+After initialization is completed, distributed training in Determined follows a loop where:
+
+#. Every worker performs a forward and backward pass on a unique mini-batch of data.
+#. As the result of the backward pass, every worker generates a set of updates to the model
+   parameters based on the data it processed.
+#. The workers communicate their updates to each other, so that all the workers see all the updates
+   made during that batch.
+#. Every worker averages the updates by the number of workers.
+#. Every worker applies the updates to its copy of the model parameters, resulting in all the
+   workers having identical solution states.
+#. Return to the first step.
+
+***********************************************
+ Reduce Computation and Communication Overhead
+***********************************************
+
+Of the steps involved in the distributed training loop in Determined, which are described above,
+step 1 and step 2 introduce the majority of the computational overhead. To reduce computational
+overhead, it's recommended that users maximize the utilization of their GPU. This is typically done
+by using the largest batch size that fits into memory. When performing distributed training, to
+reduce the computational overhead it's recommended to set the ``global_batch_size`` to the ``largest
+batch size that fits into a single GPU`` * ``number of slots``. This is commonly referred to as
+*weak scaling*.
+
+Step 3 of the distributed training loop in Determined introduces the majority of the communication
+overhead. Because deep learning models typically perform dense updates, where every model parameter
+is updated for every training sample, ``batch_size`` does not affect how long it takes workers to
+communicate updates. However, increasing ``global_batch_size`` does reduce the required number of
+passes through the training loop, thus reducing the total communication overhead.
+
+Determined optimizes the communication in step 3 by using an efficient form of ring all-reduce,
+which minimizes the amount of communication necessary for all the workers to communicate their
+updates. Determined also reduces the communication overhead by overlapping computation (step 1 &
+step 2) and communication (step 3) by communicating updates for deeper layers concurrently with
+computing updates for the shallower layers. The :ref:`multi-gpu-training` document covers additional
```

Comment:
The referenced document isn't really a thing anymore. Should it point to the implementation doc?